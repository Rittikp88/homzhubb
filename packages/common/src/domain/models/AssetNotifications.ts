import { JsonObject, JsonProperty } from '@homzhub/common/src/utils/ObjectMapper';
import { Links } from '@homzhub/common/src/domain/models/PaginationLinks';

@JsonObject('Notifications')
export class Notifications {
  @JsonProperty('id', Number)
  private _id = 0;

  @JsonProperty('notification_type', String)
  private _notificationType = '';

  @JsonProperty('title', String)
  private _title = '';

  @JsonProperty('message', String)
  private _message = '';

  @JsonProperty('link', String)
  private _link = '';

  @JsonProperty('created_at', String)
  private _createdAt = '';

  @JsonProperty('is_read', Boolean)
  private _isRead = false;

  get id(): number {
    return this._id;
  }

  get notificationType(): string {
    return this._notificationType;
  }

  get title(): string {
    return this._title;
  }

  get message(): string {
    return this._message;
  }

  get link(): string {
    return this._link;
  }

  get createdAt(): string {
    return this._createdAt;
  }

  get isRead(): boolean {
    return this._isRead;
  }
}

@JsonObject('AssetNotifications')
export class AssetNotifications {
  @JsonProperty('count', Number)
  private _count = 0;

  @JsonProperty('unread_count', Number)
  private _unreadCount = 0;

  @JsonProperty('links', Links)
  private _links: Links = new Links();

  @JsonProperty('results', [Notifications])
  private _results: Notifications[] = [];

  get count(): number {
    return this._count;
  }

  get unreadCount(): number {
    return this._unreadCount;
  }

  get links(): Links {
    return this._links;
  }

  get results(): Notifications[] {
    return this._results;
  }
}
