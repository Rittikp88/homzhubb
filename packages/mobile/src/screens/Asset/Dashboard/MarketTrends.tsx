import React from 'react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { DashboardNavigatorParamList } from '@homzhub/mobile/src/navigation/BottomTabs';
import { AnimatedProfileHeader } from '@homzhub/mobile/src/components';
import AssetMarketTrends from '@homzhub/mobile/src/components/molecules/AssetMarketTrends';

type libraryProps = NavigationScreenProps<DashboardNavigatorParamList, ScreensKeys.MarketTrends>;
type Props = WithTranslation & libraryProps;

export class MarketTrends extends React.PureComponent<Props> {
  public render = (): React.ReactElement => {
    const { t } = this.props;
    return (
      <AnimatedProfileHeader
        title={t('dashboard')}
        sectionHeader={t('marketTrends')}
        sectionTitleType="semiBold"
        onBackPress={this.handleIconPress}
      >
        <AssetMarketTrends limit={100} isHeaderVisible={false} />
      </AnimatedProfileHeader>
    );
  };

  public handleIconPress = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };
}

export default withTranslation(LocaleConstants.namespacesKey.assetDashboard)(MarketTrends);
