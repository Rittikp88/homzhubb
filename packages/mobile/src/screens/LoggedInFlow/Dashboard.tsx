import React from 'react';
import { View, StyleSheet, ScrollView } from 'react-native';
import { AssetMetricsData, AssetSubscriptionPlanData, MarketTrendsData } from '@homzhub/common/src/mocks/AssetMetrics';
import { theme } from '@homzhub/common/src/styles/theme';
import { AssetSummary } from '@homzhub/common/src/components';
import { AssetMarketTrends, AssetMetricsList, AssetSubscriptionPlan } from '@homzhub/mobile/src/components';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { LoggedInBottomTabNavigatorParamList } from '@homzhub/mobile/src/navigation/AppNavigator';

type libraryProps = NavigationScreenProps<LoggedInBottomTabNavigatorParamList, ScreensKeys.Dashboard>;
type Props = libraryProps;

export class Dashboard extends React.PureComponent<Props, {}> {
  public render = (): React.ReactElement => {
    return (
      <View style={styles.screen}>
        <ScrollView style={styles.flexOne} showsVerticalScrollIndicator={false}>
          <AssetMetricsList
            assetCount={10}
            data={AssetMetricsData}
            subscription="Homzhub Pro"
            containerStyle={styles.assetCards}
          />
          <AssetSummary notification={10} serviceTickets={20} dues={30} containerStyle={styles.assetCards} />
          <AssetMarketTrends data={MarketTrendsData} containerStyle={styles.assetCards} />
          <AssetSubscriptionPlan
            data={AssetSubscriptionPlanData}
            planName="Homzhub PRO"
            containerStyle={styles.assetCards}
          />
        </ScrollView>
      </View>
    );
  };
}

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    margin: theme.layout.screenPadding,
  },
  assetCards: {
    marginVertical: 10,
  },
  flexOne: {
    flex: 1,
  },
});
