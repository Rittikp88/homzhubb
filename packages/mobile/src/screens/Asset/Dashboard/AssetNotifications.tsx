import React from 'react';
import { StyleSheet, View } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { DashboardRepository } from '@homzhub/common/src/domain/repositories/DashboardRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { DashboardNavigatorParamList } from '@homzhub/mobile/src/navigation/BottomTabs';
import { Text, NotificationBox } from '@homzhub/common/src/components';
import { AnimatedProfileHeader, SearchBar } from '@homzhub/mobile/src/components';
import { AssetNotificationsData } from '@homzhub/common/src/mocks/AssetNotifications';

type libraryProps = NavigationScreenProps<DashboardNavigatorParamList, ScreensKeys.AssetNotifications>;
type Props = WithTranslation & libraryProps;

interface IAssetNotificationsState {
  notifications: any[];
  searchText: string;
}

export class AssetNotifications extends React.PureComponent<Props, IAssetNotificationsState> {
  public state = {
    notifications: AssetNotificationsData,
    searchText: '',
  };

  // TODO: To be uncommented when the api is ready
  // public componentDidMount = async (): Promise<void> => {
  //   await this.getAssetNotifications();
  // };

  public render = (): React.ReactNode => {
    const { t } = this.props;
    return (
      <AnimatedProfileHeader title={t('dashboard')}>
        <>
          {this.renderHeader()}
          {this.renderNotifications()}
        </>
      </AnimatedProfileHeader>
    );
  };

  public renderHeader = (): React.ReactElement => {
    const { t } = this.props;
    return (
      <View style={styles.header}>
        <Icon
          name={icons.leftArrow}
          size={20}
          color={theme.colors.primaryColor}
          onPress={this.handleIconPress}
          testID="icnBack"
        />
        <Text type="small" textType="semiBold" style={styles.headerTitle}>
          {t('notification')}
        </Text>
      </View>
    );
  };

  public renderNotifications = (): React.ReactElement => {
    const { t } = this.props;
    const { notifications, searchText } = this.state;
    return (
      <View style={styles.searchBarContainer}>
        <SearchBar
          placeholder={t('searchByKeyword')}
          value={searchText}
          updateValue={this.onUpdateSearchText}
          containerStyle={styles.searchbar}
        />
        <NotificationBox data={notifications} onPress={this.onNotificationClicked} />
      </View>
    );
  };

  public onNotificationClicked = (id: number): void => {};

  public onUpdateSearchText = (value: string): void => {
    this.setState({ searchText: value });
  };

  public handleIconPress = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };

  public getAssetNotifications = async (): Promise<void> => {
    const response = await DashboardRepository.getAssetNotifications();
    this.setState({ notifications: response });
  };
}

export default withTranslation(LocaleConstants.namespacesKey.assetDashboard)(AssetNotifications);

const styles = StyleSheet.create({
  header: {
    backgroundColor: theme.colors.white,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    padding: 10,
    borderRadius: 4,
    borderBottomRightRadius: 0,
    borderBottomLeftRadius: 0,
  },
  headerTitle: {
    color: theme.colors.darkTint1,
    marginLeft: 12,
  },
  searchbar: {
    margin: theme.layout.screenPadding,
  },
  searchBarContainer: {
    backgroundColor: theme.colors.white,
  },
});
