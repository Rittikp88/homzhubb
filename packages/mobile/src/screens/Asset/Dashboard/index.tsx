import React from 'react';
import { FlatList, TouchableOpacity, ViewStyle, TextStyle, StyleProp } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { AnalyticsService } from '@homzhub/common/src/services/Analytics/AnalyticsService';
import { ErrorUtils } from '@homzhub/common/src/utils/ErrorUtils';
import { EventType } from '@homzhub/common/src/services/Analytics/EventType';
import { AssetRepository } from '@homzhub/common/src/domain/repositories/AssetRepository';
import { DashboardRepository } from '@homzhub/common/src/domain/repositories/DashboardRepository';
import { PortfolioActions } from '@homzhub/common/src/modules/portfolio/actions';
import { RecordAssetActions } from '@homzhub/common/src/modules/recordAsset/actions';
import { UserActions } from '@homzhub/common/src/modules/user/actions';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { AssetAdvertisementBanner, AssetMetricsList, AssetSummary } from '@homzhub/mobile/src/components';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { BottomSheet } from '@homzhub/common/src/components/molecules/BottomSheet';
import AssetMarketTrends from '@homzhub/mobile/src/components/molecules/AssetMarketTrends';
import UserSubscriptionPlan from '@homzhub/common/src/components/molecules/UserSubscriptionPlan';
import FinanceOverview from '@homzhub/mobile/src/components/organisms/FinanceOverview';
import PendingPropertyListCard from '@homzhub/mobile/src/components/organisms/PendingPropertyListCard';
import { UserScreen } from '@homzhub/mobile/src/components/HOC/UserScreen';
import { Asset, PropertyStatus } from '@homzhub/common/src/domain/models/Asset';
import { AssetMetrics } from '@homzhub/common/src/domain/models/AssetMetrics';
import { Filters } from '@homzhub/common/src/domain/models/AssetFilter';
import { IActions, ISelectedAssetPlan } from '@homzhub/common/src/domain/models/AssetPlan';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { DashboardNavigatorParamList } from '@homzhub/mobile/src/navigation/DashboardStack';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { ISetAssetPayload } from '@homzhub/common/src/modules/portfolio/interfaces';
import { IApiClientError } from '@homzhub/common/src/network/ApiClientError';

interface IDispatchProps {
  setCurrentFilter: (payload: Filters) => void;
  setAssetId: (payload: number) => void;
  setSelectedPlan: (payload: ISelectedAssetPlan) => void;
  setAddPropertyFlow: (payload: boolean) => void;
  setCurrentAsset: (payload: ISetAssetPayload) => void;
  setInitialState: () => void;
}

interface IReduxStateProps {
  assets: Asset[];
}

type libraryProps = NavigationScreenProps<DashboardNavigatorParamList, ScreensKeys.DashboardLandingScreen>;
type Props = WithTranslation & libraryProps & IDispatchProps & IReduxStateProps;

interface IDashboardState {
  metrics: AssetMetrics;
  pendingProperties: Asset[];
  isLoading: boolean;
  showBottomSheet: boolean;
}

interface IFormattedBottomSheetData {
  icon: string;
  label: string;
  onPress: () => void;
}

const ShowInMvpRelease = false;

export class Dashboard extends React.PureComponent<Props, IDashboardState> {
  public focusListener: any;

  public state = {
    metrics: {} as AssetMetrics,
    pendingProperties: [],
    isLoading: false,
    showBottomSheet: false,
  };

  public componentDidMount = (): void => {
    const { navigation } = this.props;
    this.focusListener = navigation.addListener('focus', () => {
      this.getScreenData().then();
    });
  };

  public componentWillUnmount(): void {
    this.focusListener();
  }

  public render = (): React.ReactElement => {
    const { t } = this.props;
    const { isLoading, pendingProperties } = this.state;

    return (
      <UserScreen loading={isLoading} isGradient title={t('dashboard')}>
        {this.renderAssetMetricsAndUpdates()}
        {pendingProperties.length > 0 && (
          <PendingPropertyListCard
            data={pendingProperties}
            onPressComplete={this.onCompleteDetails}
            onSelectAction={this.handleActionSelection}
            onViewProperty={this.onViewProperty}
          />
        )}
        <FinanceOverview />
        <AssetMarketTrends isDashboard onViewAll={this.onViewAll} onTrendPress={this.onTrendPress} />
        <AssetAdvertisementBanner />
        {ShowInMvpRelease && <UserSubscriptionPlan onApiFailure={this.onAssetSubscriptionApiFailure} />}
        {this.renderBottomSheet()}
      </UserScreen>
    );
  };

  public renderAssetMetricsAndUpdates = (): React.ReactElement => {
    const { metrics } = this.state;
    return (
      <>
        <AssetMetricsList
          title={`${metrics?.assetMetrics?.assets?.count ?? 0}`}
          data={metrics?.assetMetrics?.miscellaneous ?? []}
          subscription={metrics?.userServicePlan?.label}
          onMetricsClicked={this.handleMetricsNavigation}
          onPlusIconClicked={this.openBottomSheet}
        />
        <AssetSummary
          notification={metrics?.updates?.notifications?.count ?? 0}
          serviceTickets={metrics?.updates?.tickets?.open?.count ?? 0}
          dues={metrics?.updates?.dues?.count ?? 0}
          messages={metrics?.updates?.messages?.unread?.count ?? 0}
          containerStyle={styles.assetCards()}
          onPressDue={this.handleDues}
          onPressServiceTickets={this.handleServiceTickets}
          onPressNotification={this.handleNotification}
          onPressMessages={this.handleMessages}
        />
      </>
    );
  };

  public renderBottomSheet = (): React.ReactElement => {
    const { t, setInitialState } = this.props;
    const { showBottomSheet } = this.state;
    const data = this.formatBottomSheetData();

    const keyExtractor = (item: IFormattedBottomSheetData, index: number): string => `${item}:${index}`;

    const renderItem = ({ item, index }: { item: IFormattedBottomSheetData; index: number }): React.ReactElement => {
      const { icon, label, onPress } = item;

      const onPressItem = (): void => {
        onPress();
        setInitialState();
        this.closeBottomSheet();
      };
      return (
        <>
          <TouchableOpacity
            style={[styles.bottomSheetItemContainer(), index % 2 === 0 && styles.evenItem()]}
            onPress={onPressItem}
          >
            <Icon name={icon} size={25} color={theme.colors.blue} />
            <Text type="small" textType="regular" style={styles.itemLabel()}>
              {label}
            </Text>
          </TouchableOpacity>
        </>
      );
    };

    return (
      <BottomSheet
        headerTitle={t('common:addCamelCase')}
        visible={showBottomSheet}
        onCloseSheet={this.closeBottomSheet}
        sheetHeight={375}
      >
        <FlatList
          data={data}
          numColumns={2}
          keyExtractor={keyExtractor}
          renderItem={renderItem}
          contentContainerStyle={styles.flatList()}
        />
      </BottomSheet>
    );
  };

  // HANDLERS
  private onAssetSubscriptionApiFailure = (err: IApiClientError): void => {
    AlertHelper.error({ message: ErrorUtils.getErrorMessage(err) });
  };

  private onCompleteDetails = (assetId: number): void => {
    const { navigation, setAssetId } = this.props;
    setAssetId(assetId);
    // @ts-ignore
    navigation.navigate(ScreensKeys.PropertyPostStack, {
      screen: ScreensKeys.AddProperty,
      params: { previousScreen: ScreensKeys.Dashboard },
    });
  };

  private onViewAll = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.MarketTrends, { isFromDashboard: true });
  };

  private onTrendPress = (url: string, trendId: number): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.WebViewScreen, { url, trendId });
  };

  private onViewProperty = (data: ISetAssetPayload): void => {
    const { setCurrentAsset, navigation } = this.props;
    setCurrentAsset(data);
    navigation.navigate(ScreensKeys.PropertyDetailScreen);
  };

  private handleMessages = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.Messages, { isFromDashboard: true });
  };

  private handleDues = (): void => {
    /**
     *
     navigation.dispatch(
     CommonActions.reset({
        index: 0,
        routes: [{ name: ScreensKeys.Financials }],
      })
     );
     */
    const { navigation, t } = this.props;
    navigation.navigate(ScreensKeys.ComingSoonScreen, { title: t('dues') ?? '', tabHeader: t('dashboard') });
  };

  private handleServiceTickets = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.ServiceTicketScreen, { isFromDashboard: true });
  };

  private handleNotification = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.AssetNotifications, { isFromDashboard: true });
  };

  private handleMetricsNavigation = (name: string): void => {
    const { navigation, setCurrentFilter } = this.props;
    setCurrentFilter(name as Filters);
    // @ts-ignore
    navigation.navigate(ScreensKeys.Portfolio, {
      screen: ScreensKeys.PortfolioLandingScreen,
    });
  };

  private handleActionSelection = (item: IActions, assetId: number): void => {
    const { navigation, setSelectedPlan, setAssetId } = this.props;
    setSelectedPlan({ id: item.id, selectedPlan: item.type });
    setAssetId(assetId);
    // @ts-ignore
    navigation.navigate(ScreensKeys.PropertyPostStack, {
      screen: ScreensKeys.AssetListing,
      params: { previousScreen: ScreensKeys.Dashboard },
    });
  };

  private openBottomSheet = (): void => this.setState({ showBottomSheet: true });

  private closeBottomSheet = (): void => this.setState({ showBottomSheet: false });

  private formatBottomSheetData = (): IFormattedBottomSheetData[] => {
    const { t, navigation, assets } = this.props;

    const handleAddProperty = (): void => {
      // @ts-ignore
      navigation.navigate(ScreensKeys.PropertyPostStack, { screen: ScreensKeys.AssetLocationSearch });
      AnalyticsService.track(EventType.AddPropertyInitiation);
    };

    const handleAddFinancialRecord = (): void => {
      if (assets.length <= 0) {
        AlertHelper.error({ message: t('assetFinancial:addProperty') });
        return;
      }
      navigation.navigate(ScreensKeys.AddRecordScreen, { isFromDashboard: true });
    };

    const handleAddServiceTicket = (): void => {
      if (assets.length <= 0) {
        AlertHelper.error({ message: t('assetDashboard:addPropertyForTicket') });
        return;
      }
      navigation.navigate(ScreensKeys.AddServiceTicket, { isFromDashboard: true });
    };

    const handleAddSupportTicket = (): void => {
      navigation.navigate(ScreensKeys.SupportScreen, { isFromDashboard: true });
    };

    const formattedDetails = [
      {
        icon: icons.portfolioFilled,
        label: t('assetFinancial:property'),
        onPress: handleAddProperty,
      },
      {
        icon: icons.barChartFilled,
        label: t('assetDashboard:incomeOrExpense'),
        onPress: handleAddFinancialRecord,
      },
      {
        icon: icons.serviceTicket,
        label: t('assetDashboard:serviceTicket'),
        onPress: handleAddServiceTicket,
      },
      {
        icon: icons.supportTicket,
        label: t('assetDashboard:supportTicket'),
        onPress: handleAddSupportTicket,
      },
    ];
    return formattedDetails;
  };

  // HANDLERS end

  // APIs
  private getScreenData = async (): Promise<void> => {
    await this.getAssetMetrics();
    await this.getPendingProperties();
  };

  private getAssetMetrics = async (): Promise<void> => {
    this.setState({ isLoading: true });
    try {
      const response: AssetMetrics = await DashboardRepository.getAssetMetrics('v3');
      this.setState({ metrics: response, isLoading: false });
    } catch (e) {
      this.setState({ isLoading: false });
      const error = ErrorUtils.getErrorMessage(e.details);
      AlertHelper.error({ message: error });
    }
  };

  private getPendingProperties = async (): Promise<void> => {
    try {
      const response: Asset[] = await AssetRepository.getPropertiesByStatus(PropertyStatus.PENDING);
      this.setState({ pendingProperties: response });
    } catch (e) {
      AlertHelper.error({ message: ErrorUtils.getErrorMessage(e.details) });
    }
  };
  // APIs end
}

export const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { setCurrentFilter, setCurrentAsset, setInitialState } = PortfolioActions;
  const { setAddPropertyFlow } = UserActions;
  const { setAssetId, setSelectedPlan } = RecordAssetActions;
  return bindActionCreators(
    { setCurrentFilter, setAssetId, setSelectedPlan, setAddPropertyFlow, setCurrentAsset, setInitialState },
    dispatch
  );
};

const mapStateToProps = (state: IState): IReduxStateProps => {
  const { getUserAssets } = UserSelector;
  return {
    assets: getUserAssets(state),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation(LocaleConstants.namespacesKey.assetDashboard)(Dashboard));

const styles = {
  assetCards: (): StyleProp<ViewStyle> => ({
    marginVertical: 12,
  }),
  evenItem: (): StyleProp<ViewStyle> => ({
    marginEnd: 17,
  }),
  bottomSheetItemContainer: (): StyleProp<ViewStyle> => ({
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 26,
    paddingVertical: 20,
    marginBottom: 20,
    borderWidth: 1,
    borderColor: theme.colors.disabled,
    borderRadius: 4,
  }),
  itemLabel: (): StyleProp<TextStyle> => ({
    marginTop: 10,
    textAlign: 'center',
    color: theme.colors.blue,
  }),
  flatList: (): StyleProp<ViewStyle> => ({
    marginBottom: 30,
    marginHorizontal: 24,
  }),
};
