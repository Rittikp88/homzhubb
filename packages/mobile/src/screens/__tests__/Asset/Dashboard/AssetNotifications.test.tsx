// @ts-nocheck
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AssetNotifications } from '@homzhub/mobile/src/screens/Asset/Dashboard/AssetNotifications';

describe('Asset Notifications Screen', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<AssetNotifications t={(key: string): string => key} />);
  });

  it('should render snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
