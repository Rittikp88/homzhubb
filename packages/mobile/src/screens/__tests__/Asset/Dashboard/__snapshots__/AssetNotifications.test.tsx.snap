// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Notifications Screen should render snapshot 1`] = `
<AnimatedProfileHeader
  title="dashboard"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#0084F8"
      name="left-arrow"
      onPress={[Function]}
      size={20}
      testID="icnBack"
    />
    <Text
      style={
        Object {
          "color": "#001B36",
          "marginLeft": 12,
        }
      }
      textType="semiBold"
      type="small"
    >
      notification
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
      }
    }
  >
    <SearchBar
      containerStyle={
        Object {
          "margin": 16,
        }
      }
      placeholder="searchByKeyword"
      updateValue={[Function]}
      value=""
    />
    <NotificationBox
      data={
        Array [
          Object {
            "description": "Computerized Tomography (CT Scan) of Left Forearm",
            "id": 1,
            "is_read": true,
            "posted_at": "2020-09-02 13:30:00",
            "project_name": "Green Ridge",
            "user": "Riki Abad",
          },
          Object {
            "description": "Bypass Left Axillary Artery to Upper Arm Vein with Autologous Venous Tissue, Open Approach",
            "id": 2,
            "is_read": false,
            "posted_at": "2020-06-26 20:15:29",
            "project_name": "Grim",
            "user": "Teresina Quartly",
          },
          Object {
            "description": "Drainage of Right Lacrimal Duct with Drainage Device, Via Natural or Artificial Opening Endoscopic",
            "id": 3,
            "is_read": false,
            "posted_at": "2019-10-25 07:03:31",
            "project_name": "4th",
            "user": "Rey Solan",
          },
          Object {
            "description": "Fusion of Occipital-cervical Joint with Synthetic Substitute, Posterior Approach, Anterior Column, Percutaneous Approach",
            "id": 4,
            "is_read": false,
            "posted_at": "2020-07-25 11:27:09",
            "project_name": "Sage",
            "user": "Kari Curm",
          },
          Object {
            "description": "Dilation of Right Internal Mammary Artery with Two Drug-eluting Intraluminal Devices, Open Approach",
            "id": 5,
            "is_read": false,
            "posted_at": "2019-12-07 01:41:01",
            "project_name": "Farwell",
            "user": "Barnett Prew",
          },
          Object {
            "description": "Extirpation of Matter from Left Lower Leg Skin, External Approach",
            "id": 6,
            "is_read": false,
            "posted_at": "2020-03-27 14:50:29",
            "project_name": "Hayes",
            "user": "Robers Oldnall",
          },
          Object {
            "description": "Excision of Right Shoulder Tendon, Open Approach, Diagnostic",
            "id": 7,
            "is_read": true,
            "posted_at": "2020-06-21 10:30:38",
            "project_name": "Kinsman",
            "user": "Sammy Maven",
          },
          Object {
            "description": "Insertion of Intraluminal Device into Right Peroneal Artery, Percutaneous Endoscopic Approach",
            "id": 8,
            "is_read": true,
            "posted_at": "2019-10-14 06:03:34",
            "project_name": "Bellgrove",
            "user": "Pearl Lambard",
          },
          Object {
            "description": "Insertion of Intraluminal Device into Left Common Iliac Vein, Percutaneous Endoscopic Approach",
            "id": 9,
            "is_read": true,
            "posted_at": "2020-06-07 13:47:35",
            "project_name": "Bayside",
            "user": "Samuele Thorogood",
          },
          Object {
            "description": "Dilation of Abdominal Aorta, Bifurcation, with Three Drug-eluting Intraluminal Devices, Percutaneous Endoscopic Approach",
            "id": 10,
            "is_read": true,
            "posted_at": "2020-01-02 21:07:56",
            "project_name": "Pennsylvania",
            "user": "Riva Ranshaw",
          },
          Object {
            "description": "Control Bleeding in Right Upper Extremity, Percutaneous Approach",
            "id": 11,
            "is_read": false,
            "posted_at": "2019-09-05 10:44:13",
            "project_name": "Ridge Oak",
            "user": "Wanda McNulty",
          },
          Object {
            "description": "Change Drainage Device in Left Upper Extremity, External Approach",
            "id": 12,
            "is_read": true,
            "posted_at": "2020-07-19 12:53:43",
            "project_name": "Gina",
            "user": "Nari Pigne",
          },
          Object {
            "description": "Repair Left Trunk Tendon, Open Approach",
            "id": 13,
            "is_read": false,
            "posted_at": "2020-06-02 03:01:52",
            "project_name": "Hauk",
            "user": "Goldie Shiel",
          },
          Object {
            "description": "Bypass Middle Esophagus to Stomach with Nonautologous Tissue Substitute, Via Natural or Artificial Opening Endoscopic",
            "id": 14,
            "is_read": true,
            "posted_at": "2019-11-29 13:51:01",
            "project_name": "Donald",
            "user": "Fredericka Brosoli",
          },
          Object {
            "description": "Bypass Right External Iliac Artery to Left External Iliac Artery with Autologous Venous Tissue, Percutaneous Endoscopic Approach",
            "id": 15,
            "is_read": false,
            "posted_at": "2020-04-30 17:49:08",
            "project_name": "Alpine",
            "user": "Lucie Onthank",
          },
          Object {
            "description": "Revision of Nonautologous Tissue Substitute in Right Auditory Ossicle, Via Natural or Artificial Opening Endoscopic",
            "id": 16,
            "is_read": false,
            "posted_at": "2020-01-25 19:51:28",
            "project_name": "Eastwood",
            "user": "Michaela Tyne",
          },
          Object {
            "description": "Reposition Left Maxilla, Open Approach",
            "id": 17,
            "is_read": true,
            "posted_at": "2020-06-09 03:52:00",
            "project_name": "Miller",
            "user": "Arel Redrup",
          },
          Object {
            "description": "Drainage of Right Tarsal, Percutaneous Endoscopic Approach, Diagnostic",
            "id": 18,
            "is_read": true,
            "posted_at": "2020-07-20 05:44:35",
            "project_name": "Anzinger",
            "user": "Coralyn Gush",
          },
          Object {
            "description": "Repair Abducens Nerve, Percutaneous Endoscopic Approach",
            "id": 19,
            "is_read": true,
            "posted_at": "2020-03-21 14:52:35",
            "project_name": "2nd",
            "user": "Gladys Simone",
          },
          Object {
            "description": "Dilation of Left Vertebral Artery, Bifurcation, with Three Drug-eluting Intraluminal Devices, Percutaneous Endoscopic Approach",
            "id": 20,
            "is_read": true,
            "posted_at": "2020-06-25 01:10:45",
            "project_name": "Farwell",
            "user": "Irina Mc Gaughey",
          },
          Object {
            "description": "Repair Right Upper Arm Skin, External Approach",
            "id": 21,
            "is_read": false,
            "posted_at": "2019-12-01 16:41:39",
            "project_name": "Union",
            "user": "Cristian Mochar",
          },
          Object {
            "description": "Removal of Synthetic Substitute from Left Ear, Percutaneous Endoscopic Approach",
            "id": 22,
            "is_read": false,
            "posted_at": "2020-05-22 14:12:32",
            "project_name": "Mcbride",
            "user": "Sofia Petran",
          },
          Object {
            "description": "Replacement of Left Tarsal with Autologous Tissue Substitute, Percutaneous Approach",
            "id": 23,
            "is_read": false,
            "posted_at": "2019-11-25 03:18:03",
            "project_name": "Trailsway",
            "user": "Odell Mates",
          },
          Object {
            "description": "Extraction of Right Sphenoid Sinus, Percutaneous Endoscopic Approach",
            "id": 24,
            "is_read": false,
            "posted_at": "2020-01-25 04:45:00",
            "project_name": "Stephen",
            "user": "Bari Duffield",
          },
          Object {
            "description": "Change Traction Apparatus on Left Lower Leg",
            "id": 25,
            "is_read": false,
            "posted_at": "2020-08-30 09:30:15",
            "project_name": "Northwestern",
            "user": "Naomi Christoffels",
          },
        ]
      }
      onPress={[Function]}
    />
  </View>
</AnimatedProfileHeader>
`;
